### PR TITLE
[CH][HOTFIX] Fix test case failed when there are some null values for the position function

### PR DESCRIPTION
**Pull request by zzcclp**
## What changes were proposed in this pull request?

Fix test case failed when there are some null values for the position function.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zzcclp** [comment]
@exmy please take a review.

File: backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetSuite.scala [hunk 1]
```diff
@@ -660,14 +660,15 @@ class GlutenClickHouseTPCHParquetSuite extends GlutenClickHouseTPCHAbstractSuite
...
       // l_returnflag: N
-      // l_linenumber: 1
       """
         |select position('D', l_shipinstruct, 0), position('', l_shipinstruct, 0),
         |position('I', l_shipinstruct, 5), position('IN', l_shipinstruct),
         |position('', l_shipinstruct), locate(l_returnflag, l_shipinstruct),
         |position(l_returnflag in l_shipinstruct), position('bar', 'foobarbar'),
         |position(l_returnflag, 'TENSTNTEST', 4), position('bar', 'foobarbar', 5),
-        |position(l_returnflag, l_shipinstruct, l_linenumber + 11)
+        |position(l_returnflag, l_shipinstruct, 12),
```
Comment (exmy) [inline backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetSuite.scala:669]: 'l_linenumber + 11' here is intended to ensure that start_pos is not a constant column

Comment (zzcclp) [reply backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetSuite.scala:669]: Done

**Comment by zzcclp** [comment]
fix in #1183 .

**Pull request closed without merging**